src/assets/RestApi/rest-api.md

```diff
@@ -0,0 +1,268 @@
+
+app = Flask(__name__)
+```
+
+3. Define the data structure for our tasks. For simplicity, we'll use a list of dictionaries:
+```python
+tasks = [
+    {"id": 1, "title": "Buy groceries", "completed": False},
+    {"id": 2, "title": "Clean the house", "completed": True}
+]
+```
+
+### GET Requests
+To retrieve all tasks, implement a route that handles GET requests:
+```python
+@app.route("/tasks", methods=["GET"])
+def get_tasks():
+    return jsonify({"tasks": tasks})
+```
+This method will return all the tasks in JSON format.
+
+To retrieve a specific task:
+```python
+@app.route("/tasks/<int:task_id>", methods=["GET"])
+def get_task(task_id):
+    task = next((item for item in tasks if item["id"] == task_id), None)
+    if task is None:
+        return jsonify({"error": "Task not found"}), 404
+    return jsonify({"task": task})
+```
+We use the id of the tasks to identify a specific task since id is unique for every tasks. This method will look for the task with provided id and return the task with the matching id. If the task with the provided id is not found, it will return the error in JSON response indicating the task was not found, with status code `404`.
+
+### POST Request
+To add a new task, implement a route that handles POST requests:
+```python
+@app.route("/tasks", methods=["POST"])
+def create_task():
+    new_task = request.get_json()
+    tasks.append(new_task)
+    return jsonify({"task": new_task}), 201
+```
+`request.get_json()` retrieves the JSON data from the request body and adds the new task to the list of tasks. After the new task has been added to the list of tasks it will return the newly created task as a JSON response with `201` status code, indicating that the task was successfully created.
+
+### PUT Request
+To update a task, implement a route that handles PUT requests:
+```python
+@app.route("/tasks/<int:task_id>", methods=["PUT"])
+def update_task(task_id):
+    task = next((item for item in tasks if item["id"] == task_id), None)
+    if task is None:
+        return jsonify({"error": "Task not found"}), 404
+    task.update(request.get_json())
+    return jsonify({"task": task})
+```
+We use the id of the task that we want to update. First, it will look whether the task with provided id exists or not. If the task doesn't exist then it will return the error in JSON response indicating the task was not found, with status code `404`. If the task exists then `task.update(request.get_json())` will update the task with the new data from the request body and returns the updated task as a JSON response.
+
+### DELETE Request
+To delete a task, implement a route that handles DELETE requests:
+```python
+@app.route("/tasks/<int:task_id>", methods=["DELETE"])
+def delete_tasks(task_id):
+    task = next((item for item in tasks if item["id"] == task_id), None)
+    if task is None:
+        return jsonify({"error": "Task not found"}), 404
+    tasks.remove(task)
+    return jsonify({"result": "Task deleted"})
```

Comment:
```suggestion
    return jsonify({"result": "Task deleted"}), <status_code>
```

Reply:
It will take the default status code for the specific methods if status code is not provided. It will explicitly set the status code if provided.